src/editor/editor.js

```diff
@@ -268,6 +270,34 @@ export default class Editor {
 		this.commands.execute( ...args );
 	}
 
+	/**
+	 * Throws `editor-wrong-element` when checked element is `<textarea>`.
+	 *
+	 * @protected
+	 * @param {HTMLElement|String} sourceElementOrData The DOM element that will be the source for the created editor
+	 * or the editor's initial data.
+	 */
+	static _assertAllowedSourceElement( sourceElementOrData ) {
```

Comment:
You added this method to the base editor. But there are editor types where textarea is a valid source element. So, in other words, this method is not that generic and should either be made more specific or moved out of here.

The other problem with it is that it does not actually check allowed elements yet. It checks only the textarea case. So not only this method is not defined in the right place, but also it's called incorrectly.

I think the easiest solution here would be to have this duplicated – there's very little code here. The only thing which needs to be DRY-ed is the error message, so we could keep it here and document in the places where we'll be firing it that it's defined here.

Reply:
I've move element check directly to `create` methods of each class. It's going to be similar to checks for initial data:
https://github.com/ckeditor/ckeditor5-editor-balloon/blob/d5c4162fe018c65604a59c5cd0d0b0f5e53c6574/src/ballooneditor.js#L209-L212